hail/src/main/scala/is/hail/expr/ir/BinarySearch.scala

```diff
@@ -10,8 +10,12 @@ class BinarySearch(mb: EmitMethodBuilder, typ: PContainer, keyOnly: Boolean) {
   val ti: TypeInfo[_] = typeToTypeInfo(elt)
 
   val (compare: CodeOrdering.F[Int], equiv: CodeOrdering.F[Boolean], findElt: EmitMethodBuilder, t: PType) = if (keyOnly) {
-    val ttype = coerce[PBaseStruct](elt)
-    require(ttype.size == 2)
+    val ttype = elt match {
+      case t: PBaseStruct =>
+        require(t.size == 2)
+        t
+      case t: PInterval => t.representation
```

Comment:
This won't correctly handle open vs. closed endpoints. I remember now how I did binary search on intervals for the partitioner, maybe we can chat in person.

Reply:
I know that they're being properly handled where this is used, though -- I check both the returned index and the returned index - 1 for containment.